build-scripts/component-common.gradle

```diff
@@ -85,33 +75,15 @@ ext.dependsOnTheMegazord = {
         }
     }
 
-    // Wire up the megazordNative configuration to the output produced from the `full-megazord` project.
     dependencies {
-        megazordNative(project("path": ":full-megazord", "configuration": "megazordNative"))
+        megazordNative project("path": ":full-megazord", "configuration": "megazordNative")
+        implementation project("path": ":full-megazord", "configuration": "libsForTests")
     }
 
     afterEvaluate {
         android.libraryVariants.all { variant ->
             def variantName = variant.name.capitalize();
             def testTask = tasks["test${variantName}UnitTest"]
-            def processTestResTask = tasks["process${variantName}UnitTestJavaRes"]
-
-            // Copy the native libmegazord to the resource dir so that it can be loaded by JNA.
-            // Note: we have to manually copy the library to the output dir ourselves.  If we simply
-            // added the megazordNative directory to sourceSets.test.resources.srcDirs, then the
-            // android gradle plugin will refuse to copy it. For details see:
-            //
-            // * https://github.com/mozilla/application-services/pull/6476#issuecomment-2537227576
-            // * https://github.com/mozilla/glean/pull/2680#issuecomment-2056627683
-            def copyNativeMegazord = tasks.register("copy${variantName}NativeMegazord", Copy) {
-                from configurations.getByName("megazordNative")
-                into processTestResTask.destinationDir
-                // Make sure to run after the process java res task, otherwise that one may
-                // overwrite our work.
-                dependsOn(processTestResTask)
-            }
-
-            testTask.dependsOn(copyNativeMegazord)
```

Comment:
To make things work on android-components we need to package a JAR file with the libraries.  A nice side benefit of this is that we can just depend on it directly rather than have th weird copy task.